homeassistant/components/voice_assistant/pipeline.py

```diff
@@ -0,0 +1,126 @@
+"""Classes for voice assistant pipelines."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any
+
+from homeassistant.components import conversation
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.util.dt import utcnow
+
+DEFAULT_TIMEOUT = 30  # seconds
+
+
+@dataclass
+class PipelineRequest:
+    """Request to start a pipeline run."""
+
+    intent_input: str | None = None
+    conversation_id: str | None = None
+
+
+class PipelineEventType(str, Enum):
+    """Event types emitted during a pipeline run."""
+
+    RUN_START = "run-start"
+    RUN_FINISH = "run-finish"
+    INTENT_START = "intent-start"
+    INTENT_FINISH = "intent-finish"
+    ERROR = "error"
+
+
+@dataclass
+class PipelineEvent:
+    """Events emitted during a pipeline run."""
+
+    type: PipelineEventType
+    data: dict[str, Any] | None = None
+    timestamp: str = field(default_factory=lambda: utcnow().isoformat())
+
+    def as_dict(self) -> dict[str, Any]:
+        """Return a dict representation of the event."""
+        return {"type": self.type, "timestamp": self.timestamp, "data": self.data or {}}
+
+
+@dataclass
+class Pipeline:
+    """A voice assistant pipeline."""
+
+    name: str
+    language: str
```

Comment:
When it needs to be set to the HA instance language.
```suggestion
    language: str | None
```